packages/relay/src/lib/eth.ts

```diff
@@ -1630,6 +1613,61 @@ export class EthImpl implements Eth {
           .then();
       }
     }
+
+    // After the try-catch process above, the `formattedTransactionId` is potentially valid in only two scenarios:
+    //   - The transaction was successfully submitted and fully processed by CN and MN.
+    //   - The transaction encountered "SDK timeout exceeded" or "Connection Dropped" errors from the SDK but still potentially reached the consensus level.
+    // In both scenarios, polling the MN is required to verify the transaction's validity before return the transaction hash to clients.
+    if (formattedTransactionId) {
+      try {
+        const contractResult = await this.mirrorNodeClient.repeatedRequest(
+          this.mirrorNodeClient.getContractResult.name,
+          [formattedTransactionId, requestDetails],
+          this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
+          requestDetails,
+        );
```

Comment:
```suggestion
    if (transactionId) {
      try {
        const formattedId = formatTransactionIdWithoutQueryParams(transactionId);
        const contractResult = await this.mirrorNodeClient.repeatedRequest(
          this.mirrorNodeClient.getContractResult.name,
          [formattedId, requestDetails],
          this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
          requestDetails,
        );
```

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229#discussion_r1834650414

Reply:
updated in this commit https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229/commits/e56d4a5e184b28021f9c0e43b0bd69b9504d7476